plugins/sso-auth/lib/apimlHandler.js

```diff
@@ -147,20 +147,14 @@ class ApimlHandler {
     if (request.body) {
       this.logger.debug(`Authenticate with body`);
       return new Promise((resolve, reject) => {
-        this.doLogin(request, sessionState).then(result=> {
-          if (result.success === true) {
-            resolve(result);
-          } else {
-            this.authenticateViaCookie(request, sessionState).then(result=> resolve(result))
-              .catch(e => reject(e));
-          }
+        this.doLogin(request, sessionState, false).then(result=> {
+          resolve(result);
         }).catch(e=> {
-          this.authenticateViaCookie(request, sessionState).then(result=> resolve(result))
-            .catch(e => reject(e));
+          Promise.resolve({success: false});
         });
       });
     } else if (request.cookies && request.cookies[TOKEN_NAME]) {
-      return this.authenticateViaCookie(request, sessionState);
+      return this.authenticateViaCookie(request, sessionState, true);
```

Comment:
My concern is the third argument because the function `authenticateViaCookie(request, sessionState)` accepts only two arguments. Is this a typo or do you have uncommitted changes for `authenticateViaCookie`?